code/src/main/java/com/onlinecontacttracing/servlets/authentication/CheckForCalendarAuthorizationServlet.java

```diff
@@ -0,0 +1,87 @@
+package com.onlinecontacttracing.authentication;
+
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import com.google.gson.Gson;
+import java.util.ArrayList;
+
+import java.util.Collections;
+import com.onlinecontacttracing.authentication.CheckForCredentials;
+
+
+import com.google.api.client.auth.oauth2.Credential;
+import com.google.api.client.googleapis.auth.oauth2.GoogleAuthorizationCodeFlow;
+import com.google.api.client.googleapis.auth.oauth2.GoogleClientSecrets;
+import com.google.api.client.googleapis.javanet.GoogleNetHttpTransport;
+import com.google.api.client.http.javanet.NetHttpTransport;
+import com.google.api.client.json.JsonFactory;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.services.gmail.Gmail;
+import com.google.api.services.gmail.GmailScopes;
+import com.google.api.services.gmail.model.Label;
+import com.google.api.services.gmail.model.ListLabelsResponse;
+import com.google.api.client.auth.oauth2.TokenResponse;
+import com.google.api.client.auth.oauth2.AuthorizationRequestUrl;
+import java.util.HashMap;
+import com.fasterxml.jackson.databind.ObjectMapper;
```

Comment:
I don't like us having this dependency - can't we get the same behavior without imporing this code from an untrusted source?

Reply:
See comment above - try to migrate to GSON